### ISSUE TITLE
List of requirements to work on JuliaBox

### ISSUE THREAD

**Issue by jlperla** [QuantEcon/lecture-source-jl#499]
See #293 

[Edit: AS] OK, here are the results. All from a fresh JuliaBox account 

#### Essential Upgrades and Installs

These are things that users can't fix in an install script (e.g., things missing from registry, permissions over install, etc.)

* `InstantiateFromURL.jl` needs the `v0.2.0` release in the registry. (but this is mainly for error handling; users won't be using this anyway). 

* `Dierckx` for amss doesn't work (`LoadError: UndefVarError: CompilerABI not defined`)

#### Nice-to-haves 

These are packages/versions that would be nice to have in the box, but not essential since users can do this setup work themselves. 

* Expectations install 
* NLsolve install 
* StatPlots + Plots bump version to v0.21.0 (important b/c of build time)
* DataFrames bump 
* Compat bump 
*

**Comment by Nosferican** [comment]
Will we need Compat after `v"1.1"`?

**Comment by jlperla** [comment]
Nope.  I don't think there is anything we would need to rely on at that point.... but I am not holding my breath for the release.

For now, the issue is that when I go `using Compat` on JuliaBox it doesn't seem to use the latest compat... and if I go `] up Compat` it also didn't seem to work.  I think this was the only problem which we do not have any workarounds for.  If you figure it out, though, post something here.

**Comment by Nosferican** [comment]
It is recommended to at least for packages and applications to use the lowest version that has the capability one needs, not to use the latest for Compat. Jeff is in charge of the 1.1 release and should be  out this year.

**Comment by arnavs** [comment]
@Nosferican The issue is that the compat on JuliaBox doesn't have the `range(start, stop, length = n)` stuff that we wrote. So lectures will fail unless it's bumped.

**Comment by Nosferican** [comment]
Should be fine in a couple weeks when Julia 1.1 comes out... I doubt we can get JuliaComputing to make a release before then and have everyone update. Could try in the Slack channel for feedback, but at least for next semester should be fine and we can drop Compat to keep things more robust.

**Comment by jlperla** [comment]
> It is recommended to at least for packages and applications to use the lowest version that has the capability one needs, not to use the latest for Compat.

That is fine, but is there no way for us to pick our own version of Compat in the meantime?  This is the issue.  Is there any workaround on something like JuliaBox.

> Should be fine in a couple weeks when Julia 1.1 comes out...

As I said, I will believe it when I see it.  Since there hasn't been any beta period with binaries, I really hope they don't rush a release.

Even after Julia 1.1. comes out, I also am not sure how long before it is supported on JuliaBox.

**Comment by jlperla** [comment]
@arnavs You say that a "Compat.jl" bump  is a nice to have.  But doesn't the lack of it break anything that is using `range` right now?  Did you find some workaround that you didn't mention, or did you manually modify the `range` when you were testing the notebooks?

**Comment by jlperla** [comment]
Split into:
- What we have no workaround for
- What we need to change to get the QELP ones working, with a workaround
- What we need to change to get the QEALP ones working, with a workaround (wait on this).
